syntax = "proto3";

package chordService;

service Chord {
	rpc find_successor(FindSuccessorRequest) returns (FindSuccessorResponse) {}
	rpc find_predecessor(FindPredecessorRequest) returns (FindPredecessorResponse) {}
	rpc notify(NotifyRequest) returns (NotifyResponse) {}
}

message FindSuccessorRequest {
	int32 id = 1;
	int32 pathlen = 2;
}

message FindSuccessorResponse {
	int32 successorId = 1;
	int32 pathlen = 2;
	string addr = 3;
}

message FindPredecessorRequest {
}

message FindPredecessorResponse {
    int32 predecessorId = 1;
    string addr = 2;
}

message NotifyRequest {
    int32 predecessorId = 1;
    string addr = 2;
}

message NotifyResponse {
    int32 result = 1;
}